crates/hir_ty/src/lower.rs

```diff
@@ -1042,6 +1043,16 @@ fn type_for_const(db: &dyn HirDatabase, def: ConstId) -> Binders<Ty> {
     Binders::new(generics.len(), Ty::from_hir(&ctx, &data.type_ref))
 }
 
+/// Build the declared type of a const param.
+fn type_for_const_param(db: &dyn HirDatabase, def: ConstParamId) -> Binders<Ty> {
```

Comment:
as mentioned above, this should return `Ty` -- a const param can't have its own parameters.